### PR TITLE
feat: add C implementation for `math/base/special/factorial2`

### LINKED ISSUE #1892: [RFC]: add C implementation for `@stdlib/math/base/special/factorial2`
### Description

This RFC proposes adding native C implementation for `math/base/special/factorial2`:
```javascript
const double stdlib_base_factorial2( const double x);
```

### Related Issues

#649 

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by performant23**
This commit if applied adds the C implementation to `@stdlib/math/base/special/factorial2` package

## Description

This pull request:

-   adds the native C implementation to `@stdlib/math/base/special/factorial2`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1892 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by performant23** [comment]
Working on fixing the C errors

File: lib/node_modules/@stdlib/math/base/special/factorial2/README.md [hunk 1]
```diff
@@ -93,6 +93,92 @@ for ( i = 0; i < values.length; i++ ) {
...
+
+<section class="usage">
+
+### Usage
+
+```c
+#include "stdlib/math/base/special/factorial2.h"
+```
+
+#### stdlib_base_factorial2( x )
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/factorial2/README.md]:
```suggestion
#### stdlib_base_factorial2( n )
```

File: lib/node_modules/@stdlib/math/base/special/factorial2/benchmark/benchmark.native.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+// MAIN //
+
+bench( pkg+'::native', opts, function benchmark( b ) {
+	var y;
+	var i;
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		y = factorial2(i % 301);
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/factorial2/benchmark/benchmark.native.js]:
```suggestion
		y = factorial2( i % 301 );
```

File: lib/node_modules/@stdlib/math/base/special/factorial2/benchmark/c/native/benchmark.c [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+/**
+* Runs a benchmark.
+*
+* @return elapsed time in seconds
+*/
+double benchmark() {
+	double elapsed;
+	double x;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/factorial2/benchmark/c/native/benchmark.c]:
```suggestion
	int32_t x;
```

File: lib/node_modules/@stdlib/math/base/special/factorial2/examples/c/example.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+#include "stdlib/math/base/special/factorial2.h"
+#include <stdio.h>
+#include <stdint.h>
+
+int main( void ) {
+	const int32_t x[] = { 1, 10, 1, 301, 302 };
+	double b;
+	int i;
+
+	for ( i = 0; i < 5; i++ ){
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/factorial2/examples/c/example.c]:
```suggestion
	const int32_t x[] = { 1, 10, 1, 301, 302 };

	double b;
	int i;
	for ( i = 0; i < 5; i++ ){
```

File: lib/node_modules/@stdlib/math/base/special/factorial2/lib/main.js [hunk 5]
```diff
@@ -63,7 +63,7 @@ function factorial2( n ) {
 	if ( isnan( n ) ) {
 		return NaN;
 	}
-	if ( n > MAX_FACTORIAL2 ) {
+	if ( n >= MAX_FACTORIAL2 ) {
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/factorial2/lib/main.js:66]: thanks for catching this!

File: lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+*
+* @example
+* double v = stdlib_base_factorial2( 3 );
+* // returns ~3
+*/
+double stdlib_base_factorial2(const int32_t n) {
+    int last;
+    double out;
+    int v;
+    int i;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c]:
```suggestion
	int32_t last;
	double out;
	int32_t v;
	int32_t i;
```

File: lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    if (stdlib_base_is_even( v )) {
+        last = 2;
+    } else {
+        last = 3;
+    }
+    out = 1;
+    for (i = v; i >= last; i -= 2) {
+        out *= i;
+    }
+    return out;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c]:
```suggestion
	if ( stdlib_base_is_nan( n ) ){
		return 0.0/0.0; // NaN
	}
	if (n >= MAX_FACTORIAL2 ){
		return STDLIB_CONSTANT_FLOAT64_PINF;
	}
	if ( n < 0 || !stdlib_base_is_integer( n )) {
		return 0.0/0.0;
	}
	v = n;
	if (v == 0 || v == 1) {
		return 1;
	}
	if (stdlib_base_is_even( v )) {
		last = 2;
	} else {
		last = 3;
	}
	out = 1;
	for (i = v; i >= last; i -= 2) {
		out *= i;
	}
	return out;
```

**Review by Pranavchiku — APPROVED** [review]
LGTM. Thanks @performant23 !

File: lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+#include "stdlib/constants/float64/pinf.h"
+#include <stdint.h>
+
+#define MAX_FACTORIAL2 301
+
+/**
+* Evaluates the double factorial of `n`.
+*
+* @param x         input value
+* @return          Double factorial
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c]:
```suggestion
* @return          double factorial
```

File: lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+*
+* @example
+* double v = stdlib_base_factorial2( 3 );
+* // returns ~3
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c]:
```suggestion
* // returns 3
```

File: lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+*/
+double stdlib_base_factorial2(const int32_t n) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c]:
```suggestion
double stdlib_base_factorial2( const int32_t n ) {
```

File: lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+	int32_t last;
+	double out;
+	int32_t v;
+	int32_t i;
+	if ( stdlib_base_is_nan( n ) ){
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c]:
```suggestion
	if ( stdlib_base_is_nan( n ) ) {
```

File: lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+	if (stdlib_base_is_even( v )) {
+		last = 2;
+	} else {
+		last = 3;
+	}
+	out = 1;
+	for (i = v; i >= last; i -= 2) {
+		out *= i;
+	}
+	return out;
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/factorial2/src/main.c]:
```suggestion
	if ( n >= MAX_FACTORIAL2 ) {
		return STDLIB_CONSTANT_FLOAT64_PINF;
	}
	if ( n < 0 || !stdlib_base_is_integer( n ) ) {
		return 0.0/0.0;
	}
	v = n;
	if ( v == 0 || v == 1 ) {
		return 1;
	}
	if ( stdlib_base_is_even( v ) ) {
		last = 2;
	} else {
		last = 3;
	}
	out = 1;
	for ( i = v; i >= last; i -= 2 ) {
		out *= i;
	}
	return out;
```

**Review by Planeshifter — APPROVED** [review]

**Pull request merged**